modules/wallet/encrypt_test.go

```diff
@@ -248,6 +249,50 @@ func TestLock(t *testing.T) {
 	}
 }
 
+// TestInitFromSeedParallelUnlock verifies that calling InitFromSeed and then
+// Unlock() in parallel results in the correct balance.
+func TestInitFromSeedParallelUnlock(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	// create a wallet with some money
+	wt, err := createWalletTester(t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer wt.closeWt()
+	seed, _, err := wt.wallet.PrimarySeed()
+	if err != nil {
+		t.Fatal(err)
+	}
+	origBal, _, _ := wt.wallet.ConfirmedBalance()
+
+	// create a blank wallet
+	dir := filepath.Join(build.TempDir(modules.WalletDir, t.Name()+"-new"), modules.WalletDir)
+	w, err := New(wt.cs, wt.tpool, dir)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// spawn an initfromseed goroutine
+	go w.InitFromSeed(crypto.TwofishKey{}, seed)
+
+	// pause for 10ms to allow the seed sweeper to start
+	time.Sleep(time.Millisecond * 10)
```

Comment:
Since the seed scanner processes blocks by subscribing to `w.cs`, you could override the `Subscribe` method to e.g. wait on a channel before sending the first block. Then start unlocking before sending down the channel.